server/server_interceptors.go

```diff
@@ -91,33 +91,22 @@ func (s *GrpcFrameworkServer) auth(ctx context.Context) (context.Context, error)
 	// Obtain token from metadata in the context
 	token, err = grpc_auth.AuthFromMD(ctx, ContextMetadataTokenKey)
 	if err != nil {
-		return nil, auditLogWarningf(codes.Unauthenticated, "Invalid or missing authentication token")
-	}
-
-	// Determine issuer
-	issuer, err := auth.TokenIssuer(token)
-	if err != nil {
-		return nil, auditLogWarningf(codes.Unauthenticated, "Unable to obtain token issuer from authorization token")
+		return nil, auditLogWarningf(codes.Unauthenticated, err, "Invalid or missing authentication token")
 	}
 
 	// Authenticate user
-	if authenticator, ok := s.config.Security.Authenticators[issuer]; ok {
-		var claims *auth.Claims
-		claims, err = authenticator.AuthenticateToken(ctx, token)
-		if err == nil {
-			// Add authorization information back into the context so that other
-			// functions can get access to this information.
-			// If this is in the context is how functions will know that security is enabled.
-			ctx = auth.ContextSaveUserInfo(ctx, &auth.UserInfo{
-				Username: authenticator.Username(claims),
-				Claims:   *claims,
-			})
-			return ctx, nil
-		} else {
-			return nil, auditLogWarningf(codes.PermissionDenied, "Unable to authenticate token")
-		}
+	claims, err := s.config.Security.Authenticators.AuthenticateToken(ctx, token)
+	if err == nil {
+		// Add authorization information back into the context so that other
+		// functions can get access to this information.
+		// If this is in the context is how functions will know that security is enabled.
+		ctx = auth.ContextSaveUserInfo(ctx, &auth.UserInfo{
+			Username: s.config.Security.Authenticators.Username(claims),
+			Claims:   *claims,
+		})
+		return ctx, nil
 	} else {
-		return nil, auditLogWarningf(codes.Unauthenticated, "%s is not a trusted issuer", issuer)
+		return nil, auditLogWarningf(codes.PermissionDenied, err, "Unable to authenticate token")
```

Comment:
Not part of your change, but feels like the error code should be `Unauthenticated` here since we could not validate the token.